filesystem/iso9660.ksy

```diff
@@ -214,60 +92,847 @@ types:
+                size: len_dir_name
+              - id: padding_field
+                doc-ref: ecma-119 9.4.6
+                doc: |
+                  Padding field is added when len_dir_name contains an odd number
+                contents: [ 0x0 ]
+                if: len_dir_name % 2 == 1
+            instances:
+              directory_records:
+                io: _root._io
+                pos: _root.sector_size * loc_ext
+                size: _root.sector_size
+                type: directory_record
+      directory_records:
+        doc: |
+          First item "." it points to it self
+          Second item ".." it points to the parent, or also to self if it the root
+        seq:
+          - id: directory_record
+            type: directory_record
+            repeat: until
+            repeat-until: _.len_dr == 0
+      directory_record:
+        doc-ref: ecma-119 9.1
+        seq:
+          - id: len_dr
+            doc-ref: ecma-119 9.1.1
+            doc: |
+              If len_dr == 0 we do not process the body
+              If len_dr >= 1 we include field len_dr in the size
+            type: u1
+          - id: body
+            type: body
+            if: len_dr > 0x0
+        types:
+          body:
+            seq:
+              - id: ext_attr_rec_len
+                doc-ref: ecma-119 9.1.2
+                type: u1
+              - id: location_of_extent
+                doc-ref: ecma-119 9.1.3
+                type: u4bi
+              - id: data_len
+                doc-ref: ecma-119 9.1.4
+                type: u4bi
+              - id: rec_date_time
+                doc-ref: ecma-119 9.1.5
+                type: datetime_short
+              - id: file_flags_multi_extent
+                doc-ref: ecma-119 9.1.6 b7
+                type: b1
+              - id: file_flags_reserved
+                doc-ref: ecma-119 9.1.6 b5+b6
+                type: b2
+              - id: file_flags_protection
+                doc-ref: ecma-119 9.1.6 b4
+                type: b1
+              - id: file_flags_record
+                doc-ref: ecma-119 9.1.6 b3
+                type: b1
+              - id: file_flags_associated_file
+                doc-ref: ecma-119 9.1.6 b2
+                type: b1
+              - id: file_flags_directory
+                doc-ref: ecma-119 9.1.6 b1
+                type: b1
+              - id: file_flags_existence
+                doc-ref: ecma-119 9.1.6 b0
+                type: b1
+              - id: file_unit_size
+                doc-ref: ecma-119 9.1.7
+                type: u1
+              - id: interleave_gap_size
+                doc-ref: ecma-119 9.1.8
+                type: u1
+              - id: vol_seq_num
+                doc-ref: ecma-119 9.1.9
+                type: u2bi
+              - id: len_fi
+                doc-ref: ecma-119 9.1.10
+                type: u1
+              - id: file_id_file
+                doc-ref: ecma-119 9.1.11
+                type: str
```

Comment:
When called from primary this has ASCII encoding.
When called from supplementary this could have UTF-8 / UFT-16 encoding.
And there are multiple types in between, even in a loop.

How can I do that? ( not really wanting to copy the whole type )

Reply:
Using parametrized types, of course.

Reply:
@SnowFlakey I don't quite understand what do you mean by "UTF-8 / UTF-16" encoding. They are completely incompatible, how would it decide on an encoding?
@KOLANICH No, you can't do that. We don't allow passing arbitrary variables into encoding.

Reply:
Unfortunately we cannot pass even expressions.
But we can do some perversions
```yaml
params:
  - id: use_utf8
    type: b1
seq:
  - id: str_a
    type: strz
    encoding: ascii
    if: not use_utf8
  - id: str_u
    type: strz
    encoding: utf-8
    if: use_utf8
instances:
  str:
    value: (use_utf ? str_u : str_a) #doesn't work for now
```